File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        # Solution 1  Simply sort and find  but that voilates the constraint of O(n)

        s =  set(nums)
        ans = 0
        for item in s:
            if (item-1) not in s:
                l = 0
                while (item+l) in s:
                    l += 1
                ans = max(l,ans)
        return ans
 
                

        

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.empty()) return 0;

        unordered_set<int> st(nums.begin(), nums.end());
        int longest = 1;

        for (int x : st) {
            if (!st.count(x - 1)) {
                int curr = x;
                int len = 1;

                while (st.count(curr + 1)) {
                    curr++;
                    len++;
                }
                longest = max(longest, len);
            }
        }
        return longest;
    }
};